stix2/sources/memory.py

```diff
@@ -0,0 +1,297 @@
+"""
+Python STIX 2.0 Memory Source/Sink
+
+Classes:
+    MemoryStore
+    MemorySink
+    MemorySource
+
+TODO: Test everything.
+
+TODO: Use deduplicate() calls only when memory corpus is dirty (been added to)
+      can save a lot of time for successive queries
+
+NOTE: Not worrying about STIX versioning. The in memory STIX data at anytime
+      will only hold one version of a STIX object. As such, when save() is called,
+      the single versions of all the STIX objects are what is written to file.
+
+"""
+
+import json
+import os
+
+from stix2 import Bundle
+from stix2.sources import DataSink, DataSource, DataStore, make_id
+from stix2validator import validate_string
+
+
+class MemoryStore(DataStore):
+    """
+    """
+    def __init__(self, stix_data=None, name="MemoryStore"):
+        """
+        Note: It doesnt make sense to create a MemoryStore by passing
+         in existing MemorySource and MemorySink because there could
+         be data concurrency issues. Just as easy to create new MemoryStore.
+        """
+        self.name = name
+        self.id = make_id()
+        self.data = {}
+
+        if stix_data:
+            if type(stix_data) == dict:
+                # stix objects are in a bundle
+                # verify STIX json data
+                r = validate_string(json.dumps(stix_data))
```

Comment:
This stringification seems silly, as stix-validator is just going to do a `json.loads` to put it back into Python structures again.  Perhaps we need a stix-validator function that works off of parsed JSON (would `schema_validate` work?  Its docstring says it needs a string, but the code clearly doesn't treat it that way.  Looks like the docstring needs an update.)

Reply:
instead of
```python
r = validate_string(json.dumps(stix_data))
```
use the following:
```python
r = validate_instance(stix_data)
```
(and similarly for the other uses of `validate_string()`)